### PR TITLE
Additional sensors for Microsemi PowerDsine PoE Switches

### PR DESCRIPTION
**Pull request by EnzoZafra**
Added power for each port, voltage and temperature sensors for microsemi powerdsine

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

**Comment by laf** [comment]
Is there no MIB for this data?

**Comment by EnzoZafra** [comment]
I found a MIB however they change the name every time a new firmware is out and the name is very generic i.e. (poe-private-mib-v2_0). I can use the MIB if that is preferred. Should I rename it?

**Comment by laf** [comment]
Sometimes the filename is not representative of the actual MIB name, open it and look for DEFINITIONS :: - The name before that will be the mib name, they might just update the filename to show version changes.

**Comment by EnzoZafra** [comment]
Yup, that is what I meant. The name inside the mib contains a version number.

**Comment by laf** [comment]
:(

Can you post the MIB?

**Comment by EnzoZafra** [comment]
[poe-private-ver-2-0.txt](https://github.com/librenms/librenms/files/1174742/poe-private-ver-2-0.txt)
I attached the .txt version of the mib here.

**Comment by laf** [comment]
Do you have a support contract with this vendor? If so could you ask them why they do this. I'd genuinely like to use MIBs for sensors as we can convert to yaml.

**Comment by EnzoZafra** [comment]
As far as I know, we don't currently have a support contract with them. I will try to voice our concerns with them, but for the mean time, should I just rename the MIBs? These sensors will be extremely helpful for us and we'd like to monitor it as soon as possible.

**Comment by laf** [comment]
Can you upload another version of the mib, let's see what the difference is

**Comment by EnzoZafra** [comment]
I could not find older versions of the MIB. I might be wrong about the MIBs changing depending on the firmware. The version number doesn't correspond with the software or firmware version of the device.

**Comment by laf** [comment]
Then I propose we switch to using the MIB and do this in yaml. Are you able to do this or would you like me to?

**Comment by EnzoZafra** [comment]
I'll give it a shot. I haven't done it in yaml yet but seems simple enough.

**Comment by laf** [comment]
Dead easy :)

**Comment by EnzoZafra** [comment]
I'm not able to read the MIB and use the OID names. I'm not sure if the problem is with the format of the MIB or if I'm doing something wrong. Can you view my code and confirm?

File: includes/definitions/discovery/microsemipdsine.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+mib: MICROSEMI-PDSINE-MIB
+modules:
+    sensors:
+        power:
+            data:
+                -
+                    oid: portTable
+                    value: portConsumptionPower
+                    num_oid: .1.3.6.1.4.1.7428.1.2.1.1.1.3.1.
```
Comment (laf) [inline includes/definitions/discovery/microsemipdsine.yaml]: From the mib, this like it should be ` .1.3.6.1.4.1.7428.1.2.1.1.1.3.`.

File: includes/definitions/discovery/microsemipdsine.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                    descr: portConsumptionPower
+                    index: 'portConsumptionPower.{{ $index }}'
+        voltage:
+            data:
+                -
+                    oid: mainPseTable
+                    value: mainVoltage
+                    num_oid: .1.3.6.1.4.1.7428.1.2.2.1.1.2.
+                    descr: System Voltage
```
Comment (laf) [inline includes/definitions/discovery/microsemipdsine.yaml]: Can this have more than one entry as the descriptions would just be the same if so. If you can have more than one then do `descr: 'System Voltage {{ $index }}'`

**Comment by laf** [comment]
@EnzoZafra Bump ^^

**Comment by EnzoZafra** [comment]
I believe there is an issue with the MIB they supply. This is what I get from 
`./discovery.php -h host -d -m sensors`:

> #### Load disco module sensors ####
SNMP[/usr/bin/snmpbulkwalk -v2c -c COMMUNITY -OeQUs -Ih -m MICROSEMI-PDSINE-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/microsemi udp:HOSTNAME:161 portTable]
Bad operator (_): At line 36 in /opt/librenms/mibs/microsemi/MICROSEMI-PDSINE-MIB
portTable: Unknown Object Identifier (Sub-id not found: (top) -> portTable)

SNMP[/usr/bin/snmpbulkwalk -v2c -c COMMUNITY -OeQUs -Ih -m MICROSEMI-PDSINE-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/microsemi udp:HOSTNAME:161 mainPseTable]
Bad operator (_): At line 36 in /opt/librenms/mibs/microsemi/MICROSEMI-PDSINE-MIB
mainPseTable: Unknown Object Identifier (Sub-id not found: (top) -> mainPseTable)

**Comment by laf** [comment]
As a quick test, delete lines 35 - 67 of the MICROSEMI-PDSINE-MIB and try again

**Comment by EnzoZafra** [comment]
> #### Load disco module sensors ####
> SNMP[/usr/bin/snmpbulkwalk -v2c -c 'Cc7Vn37E75' -OeQUs -Ih -m MICROSEMI-PDSINE-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/microsemi udp:col-poe04:161 portTable]
> Expected "(" (_): At line 125 in /opt/librenms/mibs/microsemi/MICROSEMI-PDSINE-MIB
> Should be ACCESS (pair): At line 125 in /opt/librenms/mibs/microsemi/MICROSEMI-PDSINE-MIB
> Bad parse of OBJECT-TYPE: At line 125 in /opt/librenms/mibs/microsemi/MICROSEMI-PDSINE-MIB
> portTable: Unknown Object Identifier (Sub-id not found: (top) -> portTable)
> 
> SNMP[/usr/bin/snmpbulkwalk -v2c -c 'Cc7Vn37E75' -OeQUs -Ih -m MICROSEMI-PDSINE-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/microsemi udp:col-poe04:161 mainPseTable]
> Expected "(" (_): At line 125 in /opt/librenms/mibs/microsemi/MICROSEMI-PDSINE-MIB
> Should be ACCESS (pair): At line 125 in /opt/librenms/mibs/microsemi/MICROSEMI-PDSINE-MIB
> Bad parse of OBJECT-TYPE: At line 125 in /opt/librenms/mibs/microsemi/MICROSEMI-PDSINE-MIB
> mainPseTable: Unknown Object Identifier (Sub-id not found: (top) -> mainPseTable)

Still getting an error

**Comment by laf** [comment]
Looks like we're on a losing battle with this one. Switch the changes back to php (sorry for the hassle).

**Comment by EnzoZafra** [comment]
No worries. I need to make a new PR because I messed something up on git. https://github.com/librenms/librenms/pull/7114

**Pull request closed without merging**